### PR TITLE
fix(main menu): contrast, icons, responsiveness of main menu

### LINKED ISSUE #81: [BUG] Menu contrast
**Describe the bug**
The main menu component dropdowns (in both light and dark mode) have some inconsistent and sometimes inaccessible contrast. 

**To Reproduce**
1. Go to the [main menu component in Compound](https://emulsify-ds.github.io/compound/?path=/story/molecules-menus--main)
2. Note that the menu items in the dropdowns do not have sufficient contrast

**Expected behavior**
Menu item links should have good contrast with their background.

**Screenshots**

Screen Shot 2023-01-06 at 12 40 56 AM

**Desktop (please complete the following information):**
 - Desktop browsers

### PR DESCRIPTION
**Pull request by mndonx**
## Summary

This PR fixes/implements the following **bugs/features**
See: https://emulsify-ds.github.io/compound/?path=/story/molecules-menus--main

- Main menu contrast in light & dark mode does not past testing
Screen Shot 2023-01-17 at 4 21 25 PM

- Mobile menu icon is broken 
Screen Shot 2023-01-17 at 4 21 48 PM

- Close icon should match open icon (uses a character instead of an SVG)
- Main menu in mobile has weird padding and margins
Screen Shot 2023-01-17 at 4 21 56 PM

- Mobile main menu should align right (to allow for logo)
- Main menu is not responsive (can't go from small with open items to large and have the menu work)
Screen Shot 2023-01-17 at 4 22 29 PM

Related:
- Logo is not displaying at small widths
- Social icons not displaying
Screen Shot 2023-01-17 at 4 23 06 PM

I starting fixing stuff because the bare minimum of contrast testing was not met ... but then I found some other stuff. However, the Main Menu is currently not accessible (the second level) so I think really more work will need to be put in here. I think this is just a stop-gap. 

## Documentation update (required)

If this pull request requires a change to Emulsify documentation, those changes, updates, and/or new information must accompany this pull request.

## How to review this pull request

- [ ] After recompiling, check the Main Menu component (also check the Site Header and Article content type (and full Pages) 
- [ ] Make sure the contrast is good
- [ ] Make it small to check the mobile version. Make sure that looks good (especially on Content Type and Page versions)
- [ ] Make sure all images are loading properly in Main Menu and Social Menu (as well as the Logo at small widths.)
- [ ] Make sure when going from small (with the menu open) to large, the menu continues to work

## Closing issues

Closes #81

### CODE REVIEW

File: components/02-molecules/menus/main-menu/_03-main-menu-toggle.scss [hunk 1]
```diff
@@ -2,19 +2,23 @@
...
-
+  position: absolute;
   display: inline-block;
+  margin-left: -1em;
+  margin-top: -1em;
   padding: $space;
+  width: 100vw;
   text-decoration: none;
-  width: 100%;
+  top: 1em;
```
Comment (ryanhagerty) [inline components/02-molecules/menus/main-menu/_03-main-menu-toggle.scss:12]: Let's use `rem` throughout to make it consistent

File: components/02-molecules/menus/main-menu/_00-main-menu.scss [hunk 2]
```diff
@@ -15,9 +15,18 @@ $main-menu-medium: $medium;
...
     right: 0;
     width: 100%;
+
+    @include medium {
+      left: auto;
+      overflow-y: visible;
+      position: relative;
+      top: auto;
+      right: auto;
+      width: auto;
```
Comment (ryanhagerty) [inline components/02-molecules/menus/main-menu/_00-main-menu.scss:28]: Let's adjust the ordering of items here to match the rest of the file / be in sync with how we're doing it going forward.

File: components/02-molecules/menus/main-menu/_00-main-menu.scss [hunk 3]
```diff
@@ -47,7 +56,7 @@ $main-menu-medium: $medium;
   width: 100%;
 
   @include medium {
-    background-color: clr(highlight-high);
+    background-color: clr(accent-high);
```
Comment (ryanhagerty) [inline components/02-molecules/menus/main-menu/_00-main-menu.scss:59]: Same here

**Review by ryanhagerty — CHANGES_REQUESTED** [review]
Great work, Amanda. Just a few minor changes and this should be GTG. 🏅 

Looking more closely, there are some strange choices that were preexisting. I think it's a side effect of constructing a skeleton for Compound rather than something fully fleshed out, but there's a couple areas we could take this further without dipping our toes into making this opinionated.

**Comment by mndonx** [comment]
@ryanhagerty Thank you for the review! I've adjusted the order of the CSS throughout the component and changed some ems to rems. I also swapped the `:focus` to `:focus-within` so people tabbing through the main menu in desktop can actually get to the second level. Not a perfect a11y solution, but better than what it is now!

**Review by ryanhagerty — APPROVED** [review]
LGTM - good job 🚂 

There's a couple things I noticed that I think can be in a follow up task that were pre-existing. So feel free to ignore the below, but wanted to call it out for compound as a whole:

1. there's lots of mixing of `rem` and `px`. We should go ahead make a group decision on when to apply what and where
2. Variables like `$space`. They get used sometimes, but other times I see `1rem` or even `16px`. Additionally, `$space` is used in different contexts. In my mind, `$space` should be reserved for vertical rhythm, like between components, headings, etc. **not** for borders, padding, gaps, etc. This is an example of reusing for the sake of reusing in my mind.
3. Some components like this menu rely on assumptions about the height of the component like `top: 110px`. Anyone using this component will have to refactor that, so we should make these kind of things magic number free.

Thank you for attending my unrelated rant.

**Comment by mndonx** [comment]
@ryanhagerty I totally agree, except I'm fairly sure this component is going to be completely rewritten for the UI Kit (using a very different method) and back-ported here, so I probably already spent too much time on it already -- I think it's just making sure it isn't completely broken right now. Maybe we should create an issue for fixing inconsistencies in rem/$space/px throughout the project, though? I'll do that, and set this one ready to merge.

**Review by mariannuar — APPROVED** [review]

**Comment by mcortes19** [comment]
@mndonx Could you please review/fix conflicts, thanks for all you effort 💪🏼

**Comment by mndonx** [comment]
@mcortes19 I fixed the conflicts (sorry about the delay!) so I think it is ready for merge. I'm not sure why the checks do not complete -- but it seems like that is the case on other PRs?

**Pull request merged**